yt/yt/library/s3/client.cpp

```diff
@@ -6,41 +6,81 @@
 
 #include <library/cpp/string_utils/base64/base64.h>
 
+#include <contrib/libs/poco/XML/include/Poco/XML/XML.h>
+#include <contrib/libs/poco/XML/include/Poco/DOM/AutoPtr.h>
+#include <contrib/libs/poco/XML/include/Poco/DOM/DOMParser.h>
+#include <contrib/libs/poco/XML/include/Poco/DOM/Document.h>
+#include <contrib/libs/poco/XML/include/Poco/DOM/Node.h>
+
 namespace NYT::NS3 {
 
 using namespace NConcurrency;
 using namespace NNet;
 
+namespace NPocoXml = Poco::XML;
+using TPocoXmlDocumentPtr = NPocoXml::AutoPtr<NPocoXml::Document>;
+
 ////////////////////////////////////////////////////////////////////////////////
 
-NXml::TDocument ParseXmlDocument(TSharedRef responseBody)
+NPocoXml::Node* GetRootNode(TPocoXmlDocumentPtr document)
+{
+    for (auto* child = document->firstChild(); child != nullptr; child = child->nextSibling()) {
+        // Skip the comment nodes.
+        if (child->nodeType() == NPocoXml::Node::ELEMENT_NODE) {
+            return child;
+        }
+    }
+    THROW_ERROR_EXCEPTION("The document does not have a non-comment root node");
+}
+
+//! Returns a pair <ParsedDocument, RootNode>. We need both of them as the first will manage the lifetime
+//! of the parsed document, while the second one is usually what we actually want to access.
+std::pair<TPocoXmlDocumentPtr, NPocoXml::Node*> ParseXmlDocument(TSharedRef responseBody)
```

Comment:
Let's make this a named struct. Or even a small wrapper that stores both and overrides operator* and -> to return node stuff. 

Where is an aliasing constructor when you need one :(